lib/usd/translators/shading/mtlxMaterialXSurfaceShaderWriter.cpp

```diff
@@ -897,4 +979,125 @@ void MtlxMaterialXSurfaceShaderWriter::_AddDependentNodes(
+                  { _mxColorSpaceTokens->g22_rec709, GfColorSpaceNames->G22Rec709 },
+                  { _mxColorSpaceTokens->gamma22, GfColorSpaceNames->G22Rec709 },
+                  { _mxColorSpaceTokens->g18_rec709, GfColorSpaceNames->G18Rec709 },
+                  { _mxColorSpaceTokens->gamma18, GfColorSpaceNames->G18Rec709 },
+                  { _mxColorSpaceTokens->g22_ap1, GfColorSpaceNames->G22AP1 },
+                  { _mxColorSpaceTokens->srgb_displayp3, GfColorSpaceNames->SRGBP3D65 },
+                  { _mxColorSpaceTokens->adobergb, GfColorSpaceNames->G22AdobeRGB },
+                  { _mxColorSpaceTokens->none, GfColorSpaceNames->Data },
+              };
+
+        const auto it = mtlxToUSDColorSpaceMap.find(usdColorSpace);
+        if (it != mtlxToUSDColorSpaceMap.end()) {
+            usdColorSpace = it->second;
+        }
+    }
+
+    // Please note that rec709_display and gamma24 refer to the same color space. We will use
+    // rec709_display as the name to be saved in the USD document.
+    if (usdColorSpace == _mxColorSpaceTokens->rec709_display
+        || usdColorSpace == _mxColorSpaceTokens->gamma24) {
+        // We will make an exception for rec709_display and gamma24, as we can define the color
+        // space using the API...
+        auto colorSpaceDefAPI
+            = UsdColorSpaceDefinitionAPI::Apply(prim, _mxColorSpaceTokens->rec709_display);
+
+        auto redChroma = colorSpaceDefAPI.CreateRedChromaAttr();
+        redChroma.Set(GfVec2f(0.640f, 0.330f));
+        auto greenChroma = colorSpaceDefAPI.CreateGreenChromaAttr();
+        greenChroma.Set(GfVec2f(0.300f, 0.600f));
+        auto blueChroma = colorSpaceDefAPI.CreateBlueChromaAttr();
+        blueChroma.Set(GfVec2f(0.150f, 0.060f));
+        auto whitePoint = colorSpaceDefAPI.CreateWhitePointAttr();
+        whitePoint.Set(GfVec2f(0.3127f, 0.3290f));
+        auto gamma = colorSpaceDefAPI.CreateGammaAttr();
+        gamma.Set(2.4f);
+        // No linear bias/offset
+
+        return _mxColorSpaceTokens->rec709_display;
+    }
+
+    if (!TF_VERIFY(
+            GfColorSpace::IsValid(usdColorSpace),
+            "Could not map MaterialX color space '%s' to a USD recognized color space for prim at "
+            "path '%s'.",
+            mxColorSpace.c_str(),
+            prim.GetPath().GetText())) {
+        // Currently not adding color space for which we have no definition.
+        // Going further would require OCIO integration, and USD will definitely not support LUTs,
+        // so a workaround creating a placeholder color space definition will have to be considered.
+        return {};
+    }
+
+    return usdColorSpace;
+}
+
+// Copy hierarchical color space information to USD prim using the Colorspace API schema.
+void MtlxMaterialXSurfaceShaderWriter::_SetColorSpaceAPISchemaOnPrim(
+    const UsdPrim&     prim,
+    const std::string& colorSpace)
+{
+    if (prim.HasAPI<UsdColorSpaceAPI>()) {
```

Comment:
Does this condition need to be inverted?
```suggestion
    if (!prim.HasAPI<UsdColorSpaceAPI>()) {
```

Reply:
Oh, I think I misunderstood how this works. Looks like `HasAPI<UsdColorSpaceAPI>()` checks if `UsdColorSpaceAPI::Apply()` _was already called_ on that prim, correct? 

I thought it's checking if the API is "supported" by that prim, i.e. if `UsdColorSpaceAPI::Apply()` _can be called_.